python/lsst/obs/base/formatters/filter.py

```diff
@@ -0,0 +1,151 @@
+# This file is part of obs_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("FilterFormatter",)
+
+import yaml
+
+from typing import (
+    Any,
+    Optional,
+    Type,
+)
+
+from lsst.daf.butler.formatters.file import FileFormatter
+
+
+class FilterFormatter(FileFormatter):
+    """Read and write `~lsst.afw.image.Filter` filter information."""
+
+    extension = ".yaml"
+
+    unsupportedParameters = None
+    """This formatter does not support any parameters."""
+
+    def _readFile(self, path: str, pytype: Type[Any] = None) -> Any:
+        """Read a file from the path in YAML format.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open YAML format file.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from YAML file, or None
+            if the file could not be opened.
+
+        Notes
+        -----
+        The `~yaml.UnsafeLoader` is used when parsing the YAML file.
+        """
+        try:
+            with open(path, "rb") as fd:
+                data = self._fromBytes(fd.read(), pytype)
+        except FileNotFoundError:
+            data = None
+
+        return data
+
+    def _fromBytes(self, serializedDataset: bytes, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        serializedDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Expected python type to be returned.
+
+        Returns
+        -------
+        inMemoryDataset : `object`
+            The requested data as an object, or None if the string could
+            not be read.
+        """
+        data = yaml.load(serializedDataset, Loader=yaml.FullLoader)
+
+        # This will be a simple dict so we need to convert it to
+        # the Filter type -- just needs the name
+        filter = pytype(data["canonicalName"])
```

Comment:
I recommend adding `force=True`, as is done [in the `Exposure` formatter](https://github.com/lsst/afw/blob/de602ef961473a48aa591678ed0aaa87fdc2038f/src/image/ExposureFitsReader.cc#L95). That should avoid order-of-definition problems until we change the filter system. This slightly restricts the set of valid values for `pytype`, of course.

Reply:
Yes. Thanks. That is what I was missing. This formatter is only going to be called with a pytype=lsst.afw.image.Filter so I think that's fine.